Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py

```diff
@@ -811,8 +840,174 @@ def delete_file_command(client: MsGraphClient, args):
+            display_name (Required): The display name for the permission.
+            site_name (Optional): The display name of the site. Either site_id or site_name is required.
+            site_id (Optional): The unique identifier for the site. Either site_id or site_name is required.
+
+    Returns:
+        CommandResults: The results including the new permission that was created.
+
+    """
+    app_id = args["app_id"]
+    role = argToList(args["role"])
+    display_name = args["display_name"]
+    site_id = args.get("site_id") or get_site_id_from_site_name(client, args.get("site_name"))
+
+    results = client.create_site_permission(site_id, app_id, display_name, role)
+
+    return CommandResults(
+        outputs_prefix="MsGraphFiles.SitePermission",
+        outputs_key_field="id",
+        outputs=results,
+        readable_output=tableToMarkdown("Site Permission", t=_parse_permission(results))
+    )
+
+
+def update_site_permissions_command(client: MsGraphClient, args: dict[str, str]) -> CommandResults:
+    """Updates the permissions for a SharePoint site.
+
+    Args:
+        client (MsGraphClient): The Microsoft Graph client.
+        args (dict): The command arguments.
+            permission_id (Required): The ID of the permission to update.
+            role (Required): The updated role(s) for the permission. read/write/owner.
+            site_name (Optional): The display name of the site. Either site_id or site_name is required.
+            site_id (Optional): The unique identifier for the site. Either site_id or site_name is required.
+
+    Returns:
+        CommandResults: The command results with readable output.
+
+    """
+    permission_id = args["permission_id"]
+    role = argToList(args["role"])
+    site_id = args.get("site_id") or get_site_id_from_site_name(client, args.get("site_name"))
+
+    results = client.update_site_permission(site_id, permission_id, role)
+
+    return CommandResults(
+        readable_output=(
+            f"Permission {permission_id} of site {site_id} was updated successfully with new role {results['roles']}."
+        )
+    )
+
+
+def delete_site_permission_command(client: MsGraphClient, args: dict[str, str]) -> CommandResults:
+    """Deletes a permission from a SharePoint site.
+
+    Args:
+        client (MsGraphClient): The Microsoft Graph client.
+        args (dict): The command arguments.
+            permission_id (Required): The ID of the permission to delete.
+            site_name (Optional): The display name of the site. Either site_id or site_name is required.
+            site_id (Optional): The unique identifier for the site. Either site_id or site_name is required.
+
+    Returns:
+        CommandResults: The command results with readable output.
+
+    """
+    permission_id = args["permission_id"]
+    site_id = args.get("site_id") or get_site_id_from_site_name(client, args.get("site_name"))
+
+    client.delete_site_permission(site_id, permission_id)
```

Comment:
Does the API returns nice error when the Id do not exist? If not I think we should maybe handle it ourself.

Reply:
We will always receive a 204 status code,
even if the specified permission ID does not exist in the permission list.